ast/meta.go

```diff
@@ -2,11 +2,35 @@ package ast
 
 import (
 	"fmt"
-
 	"github.com/VirusTotal/gyp/pb"
 	"github.com/golang/protobuf/proto"
 )
 
+const (
+	lowerhex = "0123456789abcdef"
+)
+
+func toASCII(s string) string {
+	ascii := make([]byte, 0)
+	for i := 0; i < len(s); i++ {
+		switch {
+		case s[i] == '\n':
+			ascii = append(ascii, `\n`...)
+		case s[i] == '\r':
+			ascii = append(ascii, `\r`...)
+		case s[i] == '\t':
+			ascii = append(ascii, `\t`...)
+		case s[i] >= 32 && s[i] < 127:
+			ascii = append(ascii, s[i])
+		default:
+			ascii = append(ascii, `\x`...)
```

Comment:
This is perfectly fine, but we could also reuse the conversion to hex already implemented in the `encoding/hex` package. For example:

`ascii = append(ascii, fmt.Sprintf("\\x%s", hex.EncodeToString([]byte{s[i]}))...)`

Reply:
I find it harder to read and more inefficient, with more functions calls involved and very little gain.